### PR TITLE
Client

### PR DESCRIPTION
**Pull request by MSusik**
Additionally, two commit from Paweł Goliński, that were pushed to the Launchpad repo, but aren't available here.

### CODE REVIEW

**Comment by MSusik** [comment]
@qiubit @bdfhjk  Please review

The daemon is not functional yet, but it serves here as an example how to implement one in future.
The requirements should be automatically installed by code produced in #4. By the time it's merged the client should be run as `python2 -m dpcs.client.main.client 'command'`

File: client/main/client.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                # TO DO - implementation depends on the way how the demon is
+                # implemented
+            },
+            "system_info": {},
+            "exit_code": exit_code,
+            "stderr_output": stderr_output
+        }
+    }
+
+    system_info = systemcheck()
```
Comment (bdfhjk) [inline client/main/client.py:51]: NIT: Wrapping the systemcheck() with a local wrapper would let us later to dynamically turn it on/off. Optional.

File: client/main/client.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        exit(1)
+
+    p = subprocess.Popen(sys.argv[1],
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE)
+    _, output = p.communicate()
+    code = p.returncode
+
+    if code != EXIT_OK:
+        # output to to, co wysyłamy
```
Comment (bdfhjk) [inline client/main/client.py]: NIT: Polish comments may be confusing to international contributors.

File: client/main/client.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+        api_description_url = SERVER_ADDRESS + "vd1/paths/"
+        try:
+            response = get(api_description_url)
+        except RequestException as e:
+            handle_request_error(
+                e, "DPCS coudln't get api description"
```
Comment (bdfhjk) [inline client/main/client.py]: NIT: Spelling mistake

File: client/main/client.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+        report = generate_report(code, output)
+
+        try:
+            response = post(crash_report_url,
+                            headers=headers,
+                            data=report)
+        except RequestException as e:
+            handle_request_error(
+                e, "DPCS coudln't post crash information"
```
Comment (bdfhjk) [inline client/main/client.py]: NIT: Spelling mistake

File: client/main/client.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            )
+            exit(3)
+
+        script = response.json()['crash_report_ack'][
+                                 'solution']['shell_script']
+
+        home = path.expanduser("~")
+        scripts_directory = path.join(home, '.local/share/dpcs/')
```
Comment (bdfhjk) [inline client/main/client.py]: Why not ~/.dpcs/ ? I think it's the most popular convention.

File: client/main/demon.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                time.sleep(10)
+            else:
+                first_sent = True
+
+            api_description_url = SERVER_ADDRESS + "vd1/paths/"
+            try:
+                response = get(api_description_url)
+            except RequestException as e:
+                self._handle_request_error(
+                    e, "DPCS coudln't get api description"
```
Comment (bdfhjk) [inline client/main/demon.py]: NIT: Spelling mistake

**Comment by bdfhjk** [comment]
Static review completed. The code have high quality. Good job :+1: 

However I can't run it directly. I've tried two approaches

marek@debian:~$ python2 -m dpcs.client.main.client 'dpkg -v'
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/marek/dpcs/client/main/client.py", line 115, in <module>
    stderr=subprocess.PIPE)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

And 

marek@debian:~/dpcs/client/main$ python client.py 
Traceback (most recent call last):
  File "client.py", line 16, in <module>
    from ...systemcheck.systemcheck import systemcheck
ValueError: Attempted relative import in non-package

Readme would be appreciated!

File: client/main/client.py (hunk 1 above)
Comment (MSusik) [inline client/main/client.py:51]: As we're not sure yet how we are going to implement system info caching, this feature is not in the scope of this PR.

**Comment by MSusik** [comment]
> Readme would be appreciated

Just please wait for the package PR. If I add a readme here, it will have to be removed with the next PR.

**Comment by bdfhjk** [comment]
@MSusik  How to quickly test the PR? Could you send me a list of instructions? 

I'll try to dynamically review it and merge before Tue 18:00 (if not please ping me to do it).

**Comment by MSusik** [comment]
@bdfhjk 
`python2 -m dpcs.client.main.client 'command'`

Then chack the directory indicated in the output

**Comment by bdfhjk** [comment]
I think we should implement the communication with user what the client is doing like we have talked yesterday. 

Additionally, would it be hard to make it transparent, so the user can use the application without knowing it's inside dpcs? And then output an information about the log uploading procedure when an application exit with crash? 

Optionally there may be a switch, like standard usage will be a transparent execution mode, and dpcs -c 'app' will capture whole output and send to the server :question: 
+

**Comment by bdfhjk** [comment]
Works really good now. Thanks! :+1:

**Pull request merged**